libs/labelbox/src/labelbox/schema/workflow/nodes.py

```diff
@@ -0,0 +1,1336 @@
+"""Node classes for Project Workflows in Labelbox."""
+
+import logging
+from typing import (
+    Dict,
+    List,
+    Any,
+    Optional,
+    Literal,
+    Union,
+    cast,
+    TYPE_CHECKING,
+)
+from pydantic import Field, field_validator, model_validator
+
+from labelbox.schema.workflow.base import BaseWorkflowNode
+from labelbox.schema.workflow.enums import (
+    WorkflowDefinitionId,
+    NodeOutput,
+    FilterField,
+    FilterOperator,
+)
+from labelbox.schema.workflow.filters import Filter, NLSearchFilter
+
+if TYPE_CHECKING:
+    from labelbox.schema.workflow.project_filter import ProjectWorkflowFilter
+
+logger = logging.getLogger(__name__)
+
+
+class InitialLabelingNode(BaseWorkflowNode):
+    """Initial labeling node. Unique, always at the beginning, one output."""
+
+    label: str = Field(default="Initial labeling task", frozen=True)
+    filter_logic: Literal["and", "or"] = Field(
+        default="and", alias="filterLogic"
+    )
```

Comment:
If this only applies to a logical node, perhaps it should be defined as frozen on the base type (or ideally only  defined on the logical node).